### ISSUE TITLE
Model cannot be loaded from your given ckpt.

### ISSUE THREAD

**Issue by HorizonWind2004** [ByteDance-Seed/Bagel#43]
_IncompatibleKeys(missing_keys=['latent_pos_embed.pos_embed', 'vit_pos_embed.pos_embe
d'], unexpected_keys=['language_model.model.norm_moe_gen.weight', 'language_model.model.layers.0.input_laye
rnorm_moe_gen.weight', 'language_model.model.layers.0.mlp_moe_gen.down_proj.weight', 'language_model.model.
layers.0.mlp_moe_gen.gate_proj.weight', 'language_model.model.layers.0.mlp_moe_gen.up_proj.weight', 'langua
ge_model.model.layers.0.post_attention_layernorm_moe_gen.weight', 'language_model.model.layers.0.self_attn.
k_norm_moe_gen.weight', 'language_model.model.layers.0.self_attn.k_proj_moe_gen.bias', 'language_model.mode
l.layers.0.self_attn.k_proj_moe_gen.weight', 'language_model.model.layers.0.self_attn.o_proj_moe_gen.weight
', 'language_model.model.layers.0.self_attn.q_norm_moe_gen.weight', 'language_model.model.layers.0.self_att
n.q_proj_moe_gen.bias', 'language_model.model.layers.0.self_attn.q_proj_moe_gen.weight', 'language_model.mo
del.layers.0.self_attn.v_proj_moe_gen.bias', 'language_model.model.layers.0.self_attn.v_proj_moe_gen.weight
', 'language_model.model.layers.1.input_layernorm_moe_gen.weight', 'language_model.model.layers.1.mlp_moe_g
en.down_proj.weight', 'language_model.model.layers.1.mlp_moe_gen.gate_proj.weight', 'language_model.model.l
ayers.1.mlp_moe_gen.up_proj.weight', 'language_model.model.layers.1.post_attention_layernorm_moe_gen.weight
', 'language_model.model.layers.1.self_attn.k_norm_moe_gen.weight', 'language_model.model.layers.1.self_att
n.k_proj_moe_gen.bias', 'language_model.model.layers.1.self_attn.k_proj_moe_gen.weight', 'language_model.mo
del.layers.1.self_attn.o_proj_moe_gen.weight', 'language_model.model.layers.1.self_attn.q_norm_moe_gen.weig
ht', 'language_model.model.layers.1.self_attn.q_proj_moe_gen.bias', 'language_model.model.layers.1.self_att
n.q_proj_moe_gen.weight', 'language_model.model.layers.1.self_attn.v_proj_moe_gen.bias', 'language_model.mo
del.layers.1.self_attn.v_proj_moe_gen.weight', 'language_model.model.layers.2.input_layernorm_moe_gen.weigh
t', 'language_model.model.layers.2.mlp_moe_gen.down_proj.weight', 'language_model.model.layers.2.mlp_moe_ge
n.gate_proj.weight', 'language_model.model.layers.2.mlp_moe_gen.up_proj.weight', 'language_model.model.laye
rs.2.post_attention_layernorm_moe_gen.weight', 'language_model.model.layers.2.self_attn.k_norm_moe_gen.weig
ht', 'language_model.model.layers.2.self_attn.k_proj_moe_gen.bias', 'language_model.model.layers.2.self_att
n.k_proj_moe_gen.weight', 'language_model.model.layers.2.self_attn.o_proj_moe_gen.weight', 'language_model.
model.layers.2.self_attn.q_norm_moe_gen.weight', 'language_model.model.layers.2.self_attn.q_proj_moe_gen.bi
as', 'language_model.model.layers.2.self_attn.q_proj_moe_gen.weight', 'language_model.model.layers.2.self_a
ttn.v_proj_moe_gen.bias', 'language_model.model.layers.2.self_attn.v_proj_moe_gen.weight', 'language_model.
model.layers.3.input_layernorm_moe_gen.weight', 'language_model.model.layers.3.mlp_moe_gen.down_proj.weight
', 'language_model.model.layers.3.mlp_moe_gen.gate_proj.weight', 'language_model.model.layers.3.mlp_moe_gen
.up_proj.weight', 'language_model.model.layers.3.post_attention_layernorm_moe_gen.weight', 'language_model.
model.layers.3.self_attn.k_norm_moe_gen.weight', 'language_model.model.layers.3.self_attn.k_proj_moe_gen.bi
as', 'language_model.model.layers.3.self_attn.k_proj_moe_gen.weight', 'language_model.model.layers.3.self_a
ttn.o_proj_moe_gen.weight', 'language_model.model.layers.3.self_attn.q_norm_moe_gen.weight', 'language_mode
l.model.layers.3.self_attn.q_proj_moe_gen.bias', 'language_model.model.layers.3.self_attn.q_proj_moe_gen.we
ight', 'language_model.model.layers.3.self_attn.v_proj_moe_gen.bias', 'language_model.model.layers.3.self_a
ttn.v_proj_moe_gen.weight', 'language_model.model.layers.4.input_layernorm_moe_gen.weight', 'language_model
.model.layers.4.mlp_moe_gen.down_proj.weight', 'language_model.model.layers.4.mlp_moe_gen.gate_proj.weight'
, 'language_model.model.layers.4.mlp_moe_gen.up_proj.weight', 'language_model.model.layers.4.post_attention
_layernorm_moe_gen.weight', 'language_model.model.layers.4.self_attn.k_norm_moe_gen.weight', 'language_mode
l.model.layers.4.self_attn.k_proj_moe_gen.bias', 'language_model.model.layers.4.self_attn.k_proj_moe_gen.we
ight', 'language_model.model.layers.4.self_attn.o_proj_moe_gen.weight', 'language_model.model.layers.4.self
_attn.q_norm_moe_gen.weight', 'language_model.model.layers.4.self_attn.q_proj_moe_gen.bias', 'language_mode
l.model.layers.4.self_attn.q_proj_moe_gen.weight', 'language_model.model.layers.4.self_attn.v_proj_moe_gen.
bias', 'language_model.model.layers.4.self_attn.v_proj_moe_gen.weight', 'language_model.model.layers.5.inpu
t_layernorm_moe_gen.weight', 'language_model.model.layers.5.mlp_moe_gen.down_proj.weight', 'language_model.
model.layers.5.mlp_moe_gen.gate_proj.weight', 'language_model.model.layers.5.mlp_moe_gen.up_proj.weight',
etc.

**Comment by causalfusion** [comment]
if you set "visual_gen = False" then those paramters used for generation won't be loaded.

**Comment by HorizonWind2004** [comment]
> if you set "visual_gen = False" then those paramters used for generation won't be loaded.

Sorry, but I set visual_gen = True XD.
Probably because something like... MoE?

**Comment by HorizonWind2004** [comment]
Training arguments TrainingArguments(visual_gen=True, visual_und=True, results_dir='.
/', checkpoint_dir='.//checkpoints', wandb_project='bagel', wandb_name='run', wandb_runid='0', wandb_resume
='allow', wandb_offline=False, global_seed=4396, auto_resume=False, resume_from='/scratch/2025_05/jixie/BAG
EL-7B-MoT', resume_model_only=True, finetune_from_ema=True, log_every=10, save_every=2000, total_steps=5000
00, warmup_steps=2000, lr_scheduler='constant', lr=0.0001, min_lr=1e-07, beta1=0.9, beta2=0.95, eps=1e-15, 
ema=0.9999, max_grad_norm=1.0, timestep_shift=1.0, mse_weight=1.0, ce_weight=1.0, ce_loss_reweighting=False
, expected_num_tokens=32768, num_replicate=1, num_shard=4, sharding_strategy='HYBRID_SHARD', backward_prefe
tch='BACKWARD_PRE', cpu_offload=False, freeze_llm=False, freeze_vit=True, freeze_vae=True, freeze_und=True,
 copy_init_moe=True, use_flex=True)

**Comment by HorizonWind2004** [comment]
And the initial training loss is... almost 11 😂

**Comment by HorizonWind2004** [comment]
@causalfusion

**Comment by Andy1621** [comment]
@milely @HorizonWind2004  Hi! Please take a look at the new commit and let me know if it resolves your issue.
